kedro/framework/cli/starters.py

```diff
@@ -766,35 +805,39 @@ def _make_cookiecutter_args_and_fetch_template(
         "extra_context": config,
     }
 
-    if checkout:
-        cookiecutter_args["checkout"] = checkout
+    kedro_version_match_starters = _kedro_and_starters_version_identical()
+
     if directory:
         cookiecutter_args["directory"] = directory
 
     tools = config["tools"]
     example_pipeline = config["example_pipeline"]
     starter_path = "git+https://github.com/kedro-org/kedro-starters.git"
+    checkout_version = version if kedro_version_match_starters else "main"
 
     if "PySpark" in tools and "Kedro Viz" in tools:
         # Use the spaceflights-pyspark-viz starter if both PySpark and Kedro Viz are chosen.
         cookiecutter_args["directory"] = "spaceflights-pyspark-viz"
         # Ensures we use the same tag version of kedro for kedro-starters
-        cookiecutter_args["checkout"] = version
+        cookiecutter_args["checkout"] = checkout_version
     elif "PySpark" in tools:
         # Use the spaceflights-pyspark starter if only PySpark is chosen.
         cookiecutter_args["directory"] = "spaceflights-pyspark"
-        cookiecutter_args["checkout"] = version
+        cookiecutter_args["checkout"] = checkout_version
     elif "Kedro Viz" in tools:
         # Use the spaceflights-pandas-viz starter if only Kedro Viz is chosen.
         cookiecutter_args["directory"] = "spaceflights-pandas-viz"
+        cookiecutter_args["checkout"] = checkout_version
     elif example_pipeline == "True":
         # Use spaceflights-pandas starter if example was selected, but PySpark or Viz wasn't
         cookiecutter_args["directory"] = "spaceflights-pandas"
-        cookiecutter_args["checkout"] = version
+        cookiecutter_args["checkout"] = checkout_version
     else:
         # Use the default template path for non PySpark, Viz or example options:
         starter_path = template_path
-
+        cookiecutter_args["checkout"] = (
```

Comment:
Related to the previous comment.

If the `checkout` argument is provided and kedro version doesn't match starters, do we indeed want to sick to `main`?

In this case, there's also a discrepancy with `_get_cookiecutter_dir` method, where we use the provided `checkout` argument.